fastlane_bot/helpers/txhelpers.py

```diff
@@ -969,13 +974,58 @@ def approve_token_for_arb_contract(self, token_address: str, approval_amount: in
                             base_gas_price=baseFee, max_priority_fee=max_priority, nonce=self.get_nonce()
                         )
                     )
+                    approve_tx = self.sign_transaction(approve_tx)
+                    self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
+                    return self.submit_transaction(approve_tx)
                 except Exception as e:
                     self.ConfigObj.logger.info(
                         f"(***2***) Error when building transaction: {e.__class__.__name__} {e}")
             else:
                 return None
-        self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
-        return self.submit_transaction(approve_tx)
 
+    async def get_layer_one_gas_fee(self, signed_transaction) -> Decimal:
+        """
+        Returns the expected layer one gas fee for a layer 2 Optimism transaction
+        :param signed_transaction: the signed ethereum TX
+
+        returns: Decimal
+            The total fee (in gas token) for the l1 gas fee
+        """
+        ethereum_base_fee = await self._get_layer_one_gas_price()
+        fixed_overhead = await self._get_layer_one_fee_overhead()
+        dynamic_overhead = await self._get_layer_one_fee_scalar()
+        zero_bytes, non_zero_bytes = count_bytes(signed_transaction["rawTransaction"])
+        tx_data_gas = zero_bytes * 4 + non_zero_bytes * 16
+        tx_total_gas = (tx_data_gas + fixed_overhead) * dynamic_overhead
+        l1_data_fee = tx_total_gas * ethereum_base_fee
+        ## Dividing by 10 ** 24 because dynamic_overhead is returned in PPM format, and to convert this from WEI format to decimal format (10 ** 18).
+        l1_data_fee = Decimal(str(l1_data_fee)) / Decimal(str(10 ** 24))
+        return l1_data_fee
+
+    async def _get_layer_one_gas_price(self):
+        """
+        Returns the layer one base fee from the Layer 2 gas oracle for Optimism blockchains
+        """
+        return self.ConfigObj.GAS_ORACLE_CONTRACT.caller.basefee()
+    async def _get_layer_one_fee_overhead(self):
+        """
+        Returns the layer one fee overhead from the Layer 2 gas oracle for Optimism blockchains
+        """
+        return self.ConfigObj.GAS_ORACLE_CONTRACT.caller.l1FeeOverhead()
+    async def _get_layer_one_fee_scalar(self):
+        """
+        Returns the layer one fee overhead from the Layer 2 gas oracle for Optimism blockchains
+        """
+        return self.ConfigObj.GAS_ORACLE_CONTRACT.caller.l1FeeScalar()
 
```

Comment:
These 3 functions are one line each; I'd move that directly to where you call them; will clear up the code

Reply:
Done